presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -152,7 +152,7 @@
     private final CounterStat timeBetweenStartsSec = new CounterStat();
 
     @GuardedBy("root")
-    private AtomicLong lastRunningQueryStartTime = new AtomicLong();
+    private AtomicLong lastRunningQueryStartTime = new AtomicLong(currentTimeMillis());
```

Comment:
can you help me understand why initializing with currentTimeMillis() is important?

Reply:
The reason for this is: internal resource groups created as and when needed, and this make sure lastRunningQueryStartTime for the newly created resource group won't be set to 0. This helps make sure we won't end up letting resource group to run more than allowed queries in case RM updates are delayed and we wait for the RM update before running more queries on the resource group.